src/recipes/index.md

```diff
@@ -0,0 +1,46 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Recipes
+
+<!-- Using a lower heading to allow expanding with more categories of recipes. Not sure what an h1 heading would contain exactly -QoS -->
+## Tooling
+
+With respect to Godot, tooling usually refers to Plugins, which are code specifically loaded during the editor and possibly at runtime which aids in
+ developing the game. There are several different approaches towards this end, with varying complexity and varying benefits.
+
+
+### Tooling with Custom Resources
+
+With godot-rust, you are able to define custom `Resource`` classes which are then available to the end user.
+
+
+### Tooling with `EditorPlugin`
+
+`EditorPlugin` types are loaded during editor and runtime and are able to access the editor as well as the scene tree. This type follows the same
+functionality that a typical `EditorPlugin` class written in GDScript would, but crucially with access to the _entire Rust ecosystem_.
+
+
+### Tooling with Engine Singletons
+
+An Engine Singleton is a class instance that is always globally available (following the Singleton Anti-pattern). However,
+it cannot access the `SceneTree` through any reliable means.
```

Comment:
Please read my comments carefully. I mentioned [before](https://github.com/godot-rust/book/pull/23#discussion_r1445076784) that calling it an "anti-pattern" without further explanation is not doing anyone a favor.

Since it's explained already inside the page, there's no need for this wording here. Just call it "pattern".